### PR TITLE
Podman bug

### PR DESCRIPTION
**Pull request by mfleader**
## Changes introduced with this PR

Fix container name generation by composing a random number generator with the factory, and generating the random name in the connector's `Deploy()`.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: internal/util/strings.go [hunk 1]
```diff
@@ -1,18 +1,17 @@
 package util
 
 import (
-	"crypto/rand"
-	"math/big"
+	//"crypto/rand"
```
Comment (jaredoconnell) [inline internal/util/strings.go:4]: Stale code comment

File: tests/common.go [hunk 2]
```diff
@@ -60,38 +60,32 @@ func InspectImage(logger log.Logger, image string) *BasicInspection {
...
+	}
+	stdoutStr := stdout.String()
+	regex := regexp.MustCompile(`.*cgroup:\[(\d+)\]`)
+	userCgroupNs = regex.ReplaceAllString(stdoutStr, "$1")
+	userCgroupNs = strings.TrimSuffix(userCgroupNs, "\n")
+
+	if err := cmd1.Wait(); err != nil {
+		logger.Errorf(err.Error())
+	}
+
```
Comment (jaredoconnell) [inline tests/common.go:88]: This is not the easiest group code to read. Comments would be helpful.

**Comment by jaredoconnell** [comment]
A subsequent run had a failure, followed by it passing again.
Do you have any idea if this was caused by the Podman deployer?

Workflow output attached.
[failure.txt](https://github.com/arcalot/arcaflow-engine-deployer-podman/files/11668689/failure.txt)

File: config.go [hunk 3]
```diff
@@ -31,13 +31,16 @@ const (
 )
 
 type Podman struct {
-	Path              string `json:"path"`
-	ContainerNameRoot string `json:"ContainerNameRoot"`
+	Path                string `json:"path"`
+	ContainerNamePrefix string `json:"containerNamePrefix"`
+	// constant prefix prepended to the randomized container name string
```
Comment (jaredoconnell) [inline config.go:36]: Is comment after the line a standard anywhere? I'm not used to that.

File: connector_test.go [hunk 4]
```diff
@@ -213,26 +230,26 @@ func TestCgroupNsByContainerName(t *testing.T) {
...
-	if err != nil {
-		t.Fatalf(err.Error())
-	}
-	t.Cleanup(func() {
-		assert.NoError(t, container1.Close())
-		assert.NoError(t, container2.Close())
-	})
+	//if err != nil {
+	//	t.Fatalf(err.Error())
+	//}
```
Comment (jaredoconnell) [inline connector_test.go:251]: Stale code

**Review by jaredoconnell — CHANGES_REQUESTED** [review]
I think it's much better now.
I also tested it dozens of times with my parallel test workload, and no failures so far!

Once the stale comments are removed, this is ready for merging.

File: connector_test.go [hunk 5]
```diff
@@ -403,29 +370,28 @@ func TestCgroupNsByNamespacePath(t *testing.T) {
...
-	} else {
-		logger.Debugf("container 1 namespace: %s, container 2 namespace: %s, they're the same!", ns1, ns2)
-		logger.Debugf("Container 2 joined the namespace via namespace path: %s", namespacePath)
-	}
+	//if ns1 != ns2 {
+	//	t.Errorf("namespace1: %s and namespace2: %s do not match, failing", ns1, ns2)
+	//} else {
+	//	logger.Debugf("container 1 namespace: %s, container 2 namespace: %s, they're the same!", ns1, ns2)
+	//	logger.Debugf("Container 2 joined the namespace via namespace path: %s", namespacePath)
+	//}
```
Comment (jaredoconnell) [inline connector_test.go:393]: Stale comments.
I will say that the old comments are more explicit than the concise assertion.

File: connector_test.go [hunk 6]
```diff
@@ -586,20 +555,13 @@ func testNetworking(t *testing.T, podmanNetworking string, containerTest string,
 		readBuffer = readOutputUntil(t, plugin, *mac)
 	}
 	logger.Infof(string(readBuffer))
-	if len(readBuffer) == 0 {
-		t.Fatalf("the container did not produce any output")
-	}
+	assert.Equals(t, len(readBuffer) > 0, true)
```
Comment (jaredoconnell) [inline connector_test.go:558]: In this case, it's losing the context that it's testing for there being an output. In this case, I would add a comment that says that it means there's no output if the length is zero.

File: connector_test.go [hunk 7]
```diff
@@ -586,20 +555,13 @@ func testNetworking(t *testing.T, podmanNetworking string, containerTest string,
...
+		assert.Equals(t, strings.Contains(string(readBuffer), *expectedOutput), true)
 	} else {
-		if strings.Contains(string(readBuffer), *ip) == false {
-			t.Fatalf("expected string not found: %s, %s found instead", *ip, string(readBuffer))
-		}
-		if strings.Contains(string(readBuffer), *mac) == false {
-			t.Fatalf("expected string not found: %s, %s found instead", *mac, string(readBuffer))
-		}
+		assert.Equals(t, strings.Contains(string(readBuffer), *ip), true)
+		assert.Equals(t, strings.Contains(string(readBuffer), *mac), true)
```
Comment (jaredoconnell) [inline connector_test.go:564]: We do have a string contains assertion: https://github.com/arcalot/go-assert/blob/9911908433a59f9fecfbdc9cd872c8e26d0ebad9/equality.go#L10

Using equals instead removes helpful info from the error message.

File: connector_test.go [hunk 8]
```diff
@@ -450,28 +416,28 @@ func TestNetworkHost(t *testing.T) {
...
-	if len(readBuffer) == 0 {
-		t.Fatal("the container did not produce any output")
-	}
+	assert.Equals(t, len(readBuffer) > 0, true)
+
 	containerOutString := string(readBuffer)
-	if strings.Contains(containerOutString, ifconfigOutStr) == false {
-		t.Fatalf("expected string not found: %s", ifconfigOutStr)
-	}
+	assert.Equals(t, strings.Contains(containerOutString, ifconfigOutStr), true)
```
Comment (jaredoconnell) [inline connector_test.go:440]: This should be switched to the contains assertion to preserve the context: https://github.com/arcalot/go-assert/blob/9911908433a59f9fecfbdc9cd872c8e26d0ebad9/equality.go#L10

**Review by jaredoconnell — CHANGES_REQUESTED** [review]
It's nice to see the changes to the test quality!
There are a few things that need to be addressed.

File: connector_test.go (hunk 7 above)
Comment (mfleader) [reply connector_test.go:564]: I think the information is expressed in the assertion statement, and printing the readBuffer to the log.

**Review by jaredoconnell — APPROVED** [review]
Much better. Thanks for making the changes!

**Review by Harshith-umesh — APPROVED** [review]
LGTM

**Pull request merged**